core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -86,6 +88,8 @@
         Setting.boolSetting("script.auto_reload_enabled", true, Property.NodeScope);
     public static final Setting<Integer> SCRIPT_MAX_SIZE_IN_BYTES =
         Setting.intSetting("script.max_size_in_bytes", 65535, Property.NodeScope);
+    public static final Setting<Integer> SCRIPT_MAX_COMPILATIONS_PER_MINUTE =
+        Setting.intSetting("script.max_compilations_per_minute", 15, Property.Dynamic, Property.NodeScope);
```

Comment:
This setting doesn't have a minimum value so people can ask allow `-4` scripts per minute. Maybe set a minimum value?